Source/WebGPU/WebGPU/Queue.h

```diff
@@ -102,6 +106,8 @@ class Queue : public WGPUQueueImpl, public ThreadSafeRefCounted<Queue> {
     HashMap<uint64_t, OnSubmittedWorkScheduledCallbacks, DefaultHash<uint64_t>, WTF::UnsignedWithZeroKeyHashTraits<uint64_t>> m_onSubmittedWorkScheduledCallbacks;
     using OnSubmittedWorkDoneCallbacks = Vector<WTF::Function<void(WGPUQueueWorkDoneStatus)>>;
     HashMap<uint64_t, OnSubmittedWorkDoneCallbacks, DefaultHash<uint64_t>, WTF::UnsignedWithZeroKeyHashTraits<uint64_t>> m_onSubmittedWorkDoneCallbacks;
+    NSMutableSet<id<MTLResource>> *m_modifiedResources;
+    NSMutableSet<TextureSliceLevel*> *m_modifiedTextures;
```

Comment:
I kind of think these should be in their own class, along with `m_blitCommandEncoder`. Is there a better refactoring we can do here, instead of having a bunch of sibling member variables with the same lifetime? If we make a new class just for these fields, we can give that class its own API (perhaps even using RAII), making it harder to accidentally not manage some state correctly.

I'm imagining something like
```
class TemporaryBlitEncoder {
public:
    void copySomethingToSomethingElse(Something, SomethingElse);
    ...
private:
    id<MTLBlitCommandEncoder> m_blitCommandEncoder { nil };
    using IThinkThereIsASpecTermForThisButIForgetWhatItIs = std::variant<RetainPtr<id<MTLResource>>, TextureSliceLevel>;
    HashSet<IThinkThereIsASpecTermForThisButIForgetWhatItIs> m_resources;
};
...
class Queue {
...
    std::optional<TemporaryBlitEncoder> m_temporaryBlitEncoder;
};
```

or something like that

Reply:
Sounds good! It appears these changes are not needed, so I removed them for now